test/core/elysia.test.ts

```diff
@@ -173,4 +173,37 @@ describe('Edge Case', () => {
 		// @ts-expect-error private property
 		expect(main.getGlobalRoutes().length).toBe(2)
 	})
+
+	describe('value returned from transform has priority over the default value from schema', () => {
```

Comment:
This new test isn't exactly related to this issue, but I added it because I could not find a test ensuring this behavior.